mbofui/app/js/controllers.js

```diff
@@ -296,6 +302,56 @@ ocellus.controller('mapController', ['$compile', '$scope', '$rootScope','$filter
     getEvents(url);
   };
 
+  $(document).on('click','.declareIntentPut', function(e){
+    $scope.intendPut($(this).attr('data-intention'), $(this).attr('data-event'),$(this).attr('data-respondent'), $(this).attr('data-intention-url')  );
+  });
+
+  $(document).on('click','.declareIntentPost', function(e){
+    $scope.intendPost($(this).attr('data-intention'), $(this).attr('data-event'));
+  });
+
+  $scope.intendPut = function (intention, targetEvent, respondent, intentionUrl) {
+    data = {
+      'intention': intention,
+      'event': targetEvent,
+      'respondent': respondent
+    };
+    Bof.IntendPut(intentionUrl, data).then(function(result) {
+      leafletData.getMap().then(function(map) {
+        map.closePopup();
+      });
+      Bof.GetIntentions('/api/intentions/?username=self').then(function(intentionsList) {
```

Comment:
/api/intentions/?username=self is used 3 times in this same file. Can we make this a constant?